atd-etl/data_model/utils/process_pdfs.py

```
@@ -0,0 +1,165 @@
+    for pixel in NEW_CR3_FORM_TEST_PIXELS:
+        rgb_pixel = page.getpixel(pixel)
+        if rgb_pixel[0] != 0 or rgb_pixel[1] != 0 or rgb_pixel[2] != 0:
+            new_cr3_form = False
+            break
+    return new_cr3_form
+
+
+def crop_and_save_diagram(page, crash_id, is_new_cr3_form, extract_dir):
+    """Crop out the crash diagram and save it to the local directory.
+
+    The diagram is saved to <extract_dir>/crash_diagrams/<crash_id>.jpeg
+
+    Args:
+        page (PIL image): the CR3 pdf page as an image
+        crash_id (int): the CRIS crash ID
+        is_new_cr3_form (bool): if the CR3 is in the 'new' format
+        extract_dir (str): the local directory in which to save the file
+
+    Returns:
+        str: diagram_full_path - the full path to the saved diagram, including it's name
+        str: diagram_filename - the name of diagram file, .e.g 12345678.jpeg
+    """
+    bbox = DIAGRAM_BBOX_PIXELS["new"] if is_new_cr3_form else DIAGRAM_BBOX_PIXELS["old"]
+    diagram_image = page.crop(bbox)
+    # todo: is it ok to swith to JPEG (as is done here) and save 75% disk space?
+    diagram_filename = f"{crash_id}.jpeg"
+    diagram_full_path = os.path.join(extract_dir, "crash_diagrams", diagram_filename)
+    diagram_image.save(diagram_full_path)
+    return diagram_full_path, diagram_filename
+
+
+def get_cr3_object_key(filename, kind):
+    """Format the S3 object to which the CR3 PDF or diagram saved"""
+    return f"{ENV}/cr3s/{kind}/{filename}"
+
+
+def process_pdf(extract_dir, filename, s3_upload, index):
+    """Handles processing of one CR3 PDF document.
+
+    Args:
+        extract_dir (str): the local path to the current extract
+        filename (str): the filename of the CR3 PDF process, e.g. 123456.pdf
+        s3_upload (bool): if the diagram and PDF should be uploaded to the S3 bucket
+        index (int): the index ID of this pdf among all PDFs being processed
+    """
+    logger.info(f"Processing {filename} ({index})")
```

if you'd like to test the error handling, you can raise an exception here:

```python
raise ValueError(f"something went wrong")
```